integration-libs/opf/base/components/opf-quick-buy/apple-pay/apple-pay.component.ts

```diff
@@ -68,31 +69,47 @@ export class ApplePayComponent implements OnInit, OnDestroy {
     );
   }
 
-  quickBuyProduct(): void {
-    this.sub = combineLatest([
+  initSingleProductTransaction(): Observable<unknown> {
+    return combineLatest([
       this.currentProductService.getProduct(),
       this.cartHandlerService.checkStableCart(),
-    ])
-      .pipe(
-        switchMap(([product, _]) =>
-          this.applePayService.start(
-            product as Product,
-            this.itemCounterService.getCounter(),
-            this.applePayDigitalWallet?.countryCode as string
-          )
-        )
+    ]).pipe(
+      take(1),
```

Comment:
Could we move `take(1)` to the end?
Operators are executed in the order they are declared. This means that data transformations and side effects occur sequentially. By placing `take*` operators at the end, you ensure that all prior operators in the chain have had the chance to act on each emitted value.

Reply:
are you sure we will get the same behavior by putting take(1) at the end?
the idea is to prevent what is above take(1) to stream additional data.
However, i think if we put concatMap instead of switchMap, we can then put take(1) at the end as we ensure that the first value will be proceed with no chance to be cancelled with switchMap.

Reply:
After discussion with @rmch91 we agreed that current shape works fine in this case.